src/AutoSelect.ml

```diff
@@ -0,0 +1,209 @@
+open GobConfig;;
+open Cil;;
+
+(*Create maps that map each function to the ones called in it and the ones calling it*)
+(* TODO rewrite with Syntacticsearch?: https://github.com/goblint/cil/blob/develop/src/ext/syntacticsearch/funcFunction.ml
```

Comment:
Would be important to mention that this only detects _statically-known_ calls and none through function pointers (through which also recursion could occur).

Also, I wouldn't rewrite it with syntactic search because its implementation is much uglier than the relatively simple visitor here.